src/components/MyAccount/NewSettings/StatusBanner.tsx

```diff
@@ -0,0 +1,50 @@
+import { Banner, Text, Link } from "@nypl/design-system-react-components"
+
+export type StatusType = "" | "failure" | "usernameFailure" | "success"
+
+type StatusBannerProps = {
+  status: StatusType
+  statusMessage: string
+}
+
+export const StatusBanner = ({ status, statusMessage }: StatusBannerProps) => {
+  const bannerContent = (
+    <div style={{ alignItems: "center" }}>
+      {status === "failure" ? (
```

Comment:
Can you pull out this conditional logic so it's not in nested ternaries, and illustrates the possibilities more clearly? Something along the lines of...
```
let statusComponent
if (status === 'success'){
  statusComponent = ...
} if (status === 'failure && statusMessage){
statusComponent = ...
else {
statusComponent = }
etc
```
potentially also with some descriptive variable names that point to when a `statusMessage` might be there or not

Reply:
Lmk if this is better– I do think, since the parent component is responsible for passing the `statusMessage`, and there's several potential parents and reasons for passing a specific message, any explanation or indication of what that message will be should also live on the parent component if that makes sense